aea/mail/base.py

```diff
@@ -270,6 +291,18 @@ def put_message(self, to: Address, sender: Address,
         self._queue.put(envelope)
 
 
+class Multiplexer:
+    """This class can handle multiple connections at once."""
+
+    def __init__(self, connections: List['Connection']):
+        """
+        Initialize the connection multiplexer.
+
+        :param connections: the connections.
+        """
+        self.connections = connections
+
+
 class MailBox(object):
     """Abstract definition of a mailbox."""
```

Comment:
I propose replacing `self._connection` with the ConnectionMultiplexer. Then change the `connect` and `disconnect` methods to call the multiplexer wherein you loop over the connection instances and establish the connections.